content/blog/_posts/2021-04-rust-custom-test-harness.md

```diff
@@ -0,0 +1,296 @@
+---
+title: How to Build a Custom Test Harness in Rust
+author:
+    name: "T.J. Telan"
+    github: "tjtelan"
+description: "Building a custom test harness in Rust is less complicated following these patterns"
+date: 2021-04-26
+slug: rust-custom-test-harness 
+url: /blog/2021/04/rust-custom-test-harness
+img: blog/images/rust-custom-harness/test-harness-social.png
+tile: blog/images/rust-custom-harness/test-harness-social.png
+twitter-card: summary_large_image
+code:
+    height: 9000
+---
+
+## Let’s talk about integration testing in Rust
+
+I ran into a problem effectively using `cargo test` in [Fluvio](https://github.com/infinyon/fluvio) for integration testing. But I learned that you aren’t stuck with the functionality of `cargo test`.
```

Comment:
I think you need to spend a little bit more time talking about why the existing amenities do not meet your needs. I'd recommend listing out:

- What are the specific testing needs you have?
- Why is the built-in testing infrastructure insufficient?

Otherwise, the way you have it is rather jarring. It feels like taking a massive leap away from the classic testing approach without really understanding why.

Reply:
This is a good point. I don't think I need to give a huge explanation, but answering those questions will be important.

> What are the specific testing needs you have?
I was hoping that I answered this question in `What do I need from integration testing?` But introducing the problem earlier (at the top) will help with the jump.

> Why is the built-in testing infrastructure insufficient?
I hope I answer this question too in the same section as above.

But to be honest, the answer to this is:

I don't know that it is insufficient. But it wasn't clear how to implement a pattern w/ libtest so that tests wouldn't need to validate their environment.

Reply:
I reworked the whole lead-in. I think it does a better job posing your questions and giving some short answers that might convince readers to continue. We'll see though.